gen/src/rfcgen.act

```diff
@@ -898,12 +916,36 @@ def gen(stmts: dict[str, Stmt]) -> list[str]:
         stmt = stmts[stmt_name]
         arg_name = stmt.argument_name
         have_children = False
-        cattrs = []
+
+        # Sort the list of attributes so that attributes with cardinality 1 are
+        # first, plus a custom order for "module" substatements so that the
+        # module layout conforms to the "typical layout" outlined in the RFC.
+        module_order = [
+            "yang-version", "namespace", "prefix", "import", "include",
+            "organization", "contact", "description", "reference", "revision"
+        ]
+
+        def _compute_priority(stmt_name: str, attr: str, card: str, parent_arg_name: ?str) -> int:
+            if attr == parent_arg_name:
+                # Argument to parent statement is always first
+                return -1
+            base = 0 if card == "1" else 1
+            if (stmt_name == "module" or stmt_name == "submodule") and attr in module_order:
+                # Insert the index as well, shifting bigger cardinality first
```

Comment:
I think this function deserves a better descriptino / comments, explaining how it works. This feels nonsensical: "shifting bigger cardinality first". wut?

Actual cardinality is one thing, I think this works on the allowed / possible granularity speced in the YANG RFC, which are IIRC four distinct cases:
- 1
- 0..1
- 0..n
- 1..n

What is "bigger cardinality" out of those? and I think it means that 1 (from above) is the "bigger cardinality"... it's just weird.

Reply:
Isn't "order" better than "priority" here?